Mention permissions API Integration (#57)
WICG/file-handling: explainer.md

For more advanced use cases, such as opening a file in an existing window or displaying a notification, applications can add a [launch event handler](https://github.com/WICG/sw-launch/blob/master/explainer.md).

## Permissions API Integration

> A `"file-handling"` permission will be defined, in order to protect the user from accidentally allowing a PWA to view files. This permission prompt may asynchronously show right after the user chooses to use a PWA capabile of handling the associated file type, to open a file, so that the permission context is more understandable and relevant. This permission prompt should also show after the PWA has an opportunity to load, and the prompt should clearly show the PWA origin, so that it is clear to the user the identity of the PWA requesting file read access for the handled file. Launched files should only be queued into the `launchQueue` after the user accepts the permission and allows the PWA to view the relevant file types. User agents may retain control over default settings and how (or if) they are exposed to the user.

Reviewer: ```suggestion
A `"file-handling"` permission will be defined to protect the user from accidentally allowing a PWA to view files. This permission prompt may asynchronously show right after the user chooses to use a PWA to open a file of any of the file types the PWA tries to associate with, so that the permission context is more understandable and relevant. This permission prompt should also show after the PWA has had an opportunity to load, and the prompt should clearly show the PWA's origin, so that the identity of the PWA requesting file read access for the handled file is clear to the user. Launched files should only be queued into the `launchQueue` after the user accepts the permission and allows the PWA to handle the relevant file types. User agents may retain control over default settings and how (or if) they are exposed to the user.
```

> A `"file-handling"` permission will be defined, in order to protect the user from accidentally allowing a PWA to view files. This permission prompt may asynchronously show right after the user chooses to use a PWA capabile of handling the associated file type, to open a file, so that the permission context is more understandable and relevant. This permission prompt should also show after the PWA has an opportunity to load, and the prompt should clearly show the PWA origin, so that it is clear to the user the identity of the PWA requesting file read access for the handled file. Launched files should only be queued into the `launchQueue` after the user accepts the permission and allows the PWA to view the relevant file types. User agents may retain control over default settings and how (or if) they are exposed to the user.

Reviewer 2: Would recommend breaking this line up into 80 chars / line to make adding review comments easier.

s/capabile/capable/

Recommend breaking up the sentence: This permission prompt should also show after the PWA has ... to become two separate sentences like:

This permission prompt should also show after the PWA has an opportunity to load.
The prompt should clearly show the PWA origin, so that it is clear to the user the identity of the PWA requesting file read access for the handled file.

Author: This file generally doesn't use 80 char/line, so I've been following this convention in my PRs.

Thanks. I've updated the PR to match the rest of these comments.

Reviewer 2: To clarify, I think you should break the current file convention and convert the file to 80 chars/line as edits come in.  However, please LMK if there's some benefit to not following 80 chars/line other than trying to match the rest of the file.

Author: Fair. I'll try using to 80 char/line in future diffs. I think I've just tended to not do so in Github since most explainers I've seen in the past tended not to, but I see that 80 char/line is used for explainers as well (like https://github.com/WICG/file-system-access/edit/main/EXPLAINER.md)

The additional security-pertinent capability that this specification provides over the [file-system-access](https://github.com/WICG/file-system-access/blob/main/EXPLAINER.md#proposed-security-models) API is the ability to grant access to certain files through the operating system UI, as opposed to through a file picker shown by a web application. Any restrictions as to the files and folders that can be opened via the picker will also be applied to the files and folders opened via the operating system.

> There is still a risk that users may unintentionally grant a web application access to a file by opening it. However, it is generally understood that opening a file allows the application it is opened with to read and/or manipulate that file. Therefore, a user's explicit choice to open a file in an installed application, such as via an “Open with...” context menu, can be read as a signal of trust in the application, and in conjunction with a permission prompt, can be viewed as sufficient trust let the web application view the file.

Reviewer 2: I think this sentence should be shorter to be easier to read.  Proposal:

Therefore, a user's explicit choice to open a file in an installed application, such as via an “Open with...” context menu, can be read as a signal of trust in the application.  In conjunction with a permission prompt, this can be viewed as sufficient trust to let the web application view the file.

(note I added "to" between "trust" and "let")

Author: Oof thanks for the catch... looks like I didn't sufficiently check my grammar. Done.
